src/hepattn/models/matcher.py

```diff
@@ -6,27 +6,31 @@
 import torch
 from torch import nn
 
-import lap1015
+from hepattn.utils.import_utils import check_import_safe
 
 
 def solve_scipy(cost):
     _, col_idx = scipy.optimize.linear_sum_assignment(cost)
     return col_idx
 
 
-def solve_1015_early(cost):
-    return lap1015.lap_early(cost)
+SOLVERS = {
+    "scipy": solve_scipy,
+}
 
+# Some compiled extension can cause SIGKILL errors if compiled for the wrong arch
+# So we have to check they won't kill everything when we import them
+if check_import_safe("lap1015"):
```

Comment:
can we raise a warning that `lap1015` will be disabled in the `else` case, and suggest that the user reinstalls the python package on their new arch?